src/docfx/build/page/BuildPage.cs

```diff
@@ -98,7 +98,14 @@ private static (List<Error> errors, Schema schema, PageModel model, JObject meta
             var errors = new List<Error>();
             var content = file.ReadText();
             GitUtility.CheckMergeConflictMarker(content, file.FilePath);
-            var (html, markup) = Markup.ToHtml(content, file, xrefMap, dependencies, bookmarkValidator, buildChild, MarkdownPipelineType.ConceptualMarkdown);
+            var callback = new AttributeTransformerCallback(xrefMap, dependencies, bookmarkValidator, buildChild);
+            var (html, markup) = Markup.ToHtml(
+                content,
+                file,
+                (path, relativeTo) => AttributeTransformer.ReadFile(path, relativeTo, dependencies, errors),
```

Comment:
>AttributeTransformer.ReadFile [](start = 38, length = 29)

Move these methods to another class? seems strage for markdown pipeline to use **attribute**transformer which is an sdp module

Reply:
Move ReadFile and GetLink to Resolve.cs, but it still needs `attributetransformer` to get callbacks